src/Interpreters/InterpreterCreateQuery.cpp

```diff
@@ -1445,6 +1446,13 @@ BlockIO InterpreterCreateQuery::fillTableIfNeeded(const ASTCreateQuery & create)
         return InterpreterInsertQuery(insert, getContext(),
             getContext()->getSettingsRef().insert_allow_materialized_columns).execute();
     }
+    else if (create.select && !create.attach && create.is_window_view && create.is_populate)
```

Comment:
may be adjust it do it can be done in the same place as for mat view?
and for exception `POPULATE is not supported when using function now() as the time column` we can add some method like `isPopulateAllowed`

Reply:
I mean through the same InterpreterInsertQuery, without separate populate() method in window view storage.

Reply:
I have thought of doing this in `InterpreterInsertQuery` and do not modify `InterpreterCreateQuery`, just like `INSERT INTO WindowView`. But WindowView is a little bit different from Materialized view. `writeIntoWindowView` needs the **original** data from the source table and `write` into Materialized View needs the **calculated** data. So we cannot use the same `INSERT SELECT` query with MaterializedView in function `fillTableIfNeeded()`.

I use a `populate()` in WindowView because some variables of WindowView are needed to be used to construct the SELECT query, and we can only execute the SELECT query to stage FetchColumns as a trick.

Or may be we can create another `INSERT SELECT` query in `fillTableIfNeeded`(`InterpreterCreateQuery`)  and then insert the source table data into WindowView like a normal table?

Reply:
Yes, the code has already updated

Reply:
great, sorry I did not check